test/file.jl

```diff
@@ -1156,3 +1156,70 @@ if !Sys.iswindows()
         test_22566()
     end
 end  # !Sys.iswindows
+
+function file_prefix_test(path, prefix, pfxlen=length(prefix), fails=false)
+    if Sys.iswindows() && (pfxlen > 3)
+        pfxlen = 3
+    end
+
+    filename = basename(path)
+    filepart = filename[1:pfxlen]
+    prefixpart = prefix[1:pfxlen]
+
+    if !fails
+        @test filepart == prefixpart
+    else
+        @test filepart != prefixpart
+    end
+end
+
+function test_22922()
+    def_prefix="jl_"
+    tst_prefix="ABCDEF"
+    mktempdir() do tmpdir
+        file_prefix_test(tmpdir, def_prefix)
+    end
+    mktempdir(; prefix=tst_prefix) do tmpdir
+        file_prefix_test(tmpdir, tst_prefix)
+    end
+    mktemp() do tmp,io
+        file_prefix_test(tmp, def_prefix)
+    end
+    mktemp(; prefix=tst_prefix) do tmp,io
+        file_prefix_test(tmp, tst_prefix)
+    end
+    file_prefix_test(tempname(), def_prefix)
+    #Unix like OS have prefix size limit of 5 bytes
+    file_prefix_test(tempname(; prefix=tst_prefix), tst_prefix, 5)
+
+    #Special character prefix tests
+    tst_prefix="#!@%^&*()"
+    mktempdir(; prefix=tst_prefix) do tmpdir
+        file_prefix_test(tmpdir, tst_prefix)
+    end
+    mktemp(; prefix=tst_prefix) do tmp,io
+        file_prefix_test(tmp, tst_prefix)
+    end
+
+    #Unicode test
+    tst_prefix="\u2200x\u2203y"
+    mktempdir(; prefix=tst_prefix) do tmpdir
+        file_prefix_test(tmpdir, tst_prefix)
+    end
+    mktemp(; prefix=tst_prefix) do tmp,io
+        file_prefix_test(tmp, tst_prefix)
+    end
+    #In unix the prefix is computed upto 5 bytes and not chars so will
+    #fail on unicode test. But extracting first 5 bytes and creating an
+    #invalid 5-byte character array and converting to a string will pass
+    if !Sys.iswindows()
+        bytepfx=transcode(UInt8, tst_prefix)
```

Comment:
I think we should be consistent with whatever `mkdir` does. AFAIK it's not our job to check the contents of the string, and especially not of the prefix, which is unlikely to be a user-provided string. The OS will raise an error if the character is not allowed.

Whether to automatically truncate the prefix to 3 characters is a tough question. On the one hand it would allow a more consistent behavior. On the other hand, if a OS turns out not to support prefixes, or prefixes of only one characters, we will have to break API to ensure consistency, which is certainly not a good idea. So I think I'd rather keep the current behavior where it's unspecified whether the prefix will actually be used. Anyway these names are unpredictable for the caller, so variations are not an issue.